en/lessons/basics/basics.md

```diff
@@ -9,269 +9,197 @@ Getting started, basic data types, and basic operations.
+"*"
 ```
 
-Elixir also includes more complex data types.
-We'll learn more about these when we learn about [collections](../collections/) and [functions](../functions/).
+You'll learn more about data types in [collections](../collections/) and
+[functions](../functions/).
 
 ## Basic Operations
 
 ### Arithmetic
 
-Elixir supports the basic operators `+`, `-`, `*`, and `/` as you would expect.
-It's important to remember that `/` will always return a float:
+Haskell supports the basic operators `+`, `-`, `*`, and `/`.
+Note that `/` never returns an integer.
 
-```elixir
-iex> 2 + 2
+```haskell
+ghci> 2 + 2
 4
-iex> 2 - 1
+ghci> 2 - 1
 1
-iex> 2 * 5
+ghci> 2 * 5
 10
-iex> 10 / 5
+ghci> 10 / 5
 2.0
 ```
 
-If you need integer division or the division remainder (i.e., modulo), Elixir comes with two helpful functions to achieve this:
+For integer division and integer modulus, use `div` and `mod`:
 
-```elixir
-iex> div(10, 5)
-2
-iex> rem(10, 3)
-1
+```haskell
+ghci> -8 `div` 3
+-2
+ghci> 243 `mod` 5
+3
+ghci> 2 * (5 `div` 2) + (5 `mod` 2)
+5
 ```
 
 ### Boolean
 
-Elixir provides the `||`, `&&`, and `!` boolean operators.
-These support any types:
+Haskell provides the boolean operators `||`, `&&`, and `not` that operate on
+`Bool`:
 
-```elixir
-iex> -20 || true
--20
-iex> false || 42
-42
+```haskell
+ghci> False || False
+False
+ghci> True || False
+True
 
-iex> 42 && true
-true
-iex> 42 && nil
-nil
-
-iex> !42
-false
-iex> !false
-true
-```
+ghci> False && True
+False
+ghci> True && True
+True
 
-There are three additional operators whose first argument _must_ be a boolean (`true` or `false`):
-
-```elixir
-iex> true and 42
-42
-iex> false or true
-true
-iex> not false
-true
-iex> 42 and true
-** (ArgumentError) argument error: 42
-iex> not 42
-** (ArgumentError) argument error
+ghci> not True
+False
+ghci> not False
+True
 ```
 
-Note: Elixir's `and` and `or` actually map to `andalso` and `orelse` in Erlang.
-
 ### Comparison
 
-Elixir comes with all the comparison operators we're used to: `==`, `!=`, `===`, `!==`, `<=`, `>=`, `<`, and `>`.
-
-```elixir
-iex> 1 > 2
-false
-iex> 1 != 2
-true
-iex> 2 == 2
-true
-iex> 2 <= 3
-true
-```
-
-For strict comparison of integers and floats, use `===`:
-
-```elixir
-iex> 2 == 2.0
-true
-iex> 2 === 2.0
-false
-```
-
-An important feature of Elixir is that any two types can be compared; this is particularly useful in sorting. We don't need to memorize the sort order, but it is important to be aware of it:
-
-```elixir
-number < atom < reference < function < port < pid < tuple < map < list < bitstring
-```
-
-This can lead to some interesting, yet valid comparisons you may not find in other languages:
-
-```elixir
-iex> :hello > 999
-true
-iex> {:hello, :world} > [1, 2, 3]
-false
-```
-
-### String Interpolation
+Haskell comes with comparison operators for ordering: `<=`, `>=`, `<`, and `>`.
+The operator `==` compares equality whilst `/=` compares inequality.
+Values compared must be of the same type.
```

Comment:
These sentences could have a bit more cohesion. The introduction of equality and inequality seems abrupt.

Reply:
> The operator `==` compares equality whilst `/=` compares inequality.

I'd consider rewording this. While I can catch the text's drift, a literal reading makes it sound equalities are being compared, rather than two values being compared to test for equality.

Maybe rewrite to:
> The operator `==` tests for equality whilst `/=` tests for inequality.
or
> The operator `==` checks for equality whilst `/=` checks for inequality.
---
> Values compared must be of the same type.
The phrasing seems a little bit awkward there. Consider:
> (The) compared values must have the same type.